packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/Mocks.h

```diff
@@ -0,0 +1,127 @@
+@property(nonatomic, strong, nonnull)
+    SKPaymentQueue *realQueue;  // This is a reference to the real SKPaymentQueue
+
+// Stubs
+@property(nonatomic, copy, nullable) void (^showPriceConsentIfNeededStub)(void);
+@property(nonatomic, copy, nullable) void (^restoreTransactionsStub)(NSString *);
+@property(nonatomic, copy, nullable) void (^startObservingPaymentQueueStub)(void);
+@property(nonatomic, copy, nullable) void (^stopObservingPaymentQueueStub)(void);
+@property(nonatomic, copy, nullable) void (^presentCodeRedemptionSheetStub)(void);
+@property(nonatomic, copy, nullable)
+    NSArray<SKPaymentTransaction *> * (^getUnfinishedTransactionsStub)(void);
+
+@end
+
+// An interface representing a stubbed DefaultTransactionCache
+@interface TransactionCacheStub : NSObject <FLTTransactionCacheProtocol>
+
+// Stubs
+@property(nonatomic, copy, nullable) NSArray * (^getObjectsForKeyStub)(TransactionCacheKey key);
+@property(nonatomic, copy, nullable) void (^clearStub)(void);
+@property(nonatomic, copy, nullable) void (^addObjectsStub)(NSArray *, TransactionCacheKey);
+
+@end
+
+// An interface representing a stubbed DefaultMethodChannel
+@interface MethodChannelStub : NSObject <FLTMethodChannelProtocol>
+
+// Stubs
+@property(nonatomic, copy, nullable) void (^invokeMethodChannelStub)(NSString *method, id arguments)
+    ;
+@property(nonatomic, copy, nullable) void (^invokeMethodChannelWithResultsStub)
+    (NSString *method, id arguments, FlutterResult _Nullable);
+
+@end
+
+// An interface representing a stubbed DefaultPaymentQueueHandler
+@interface PaymentQueueHandlerStub
+    : NSObject <SKPaymentTransactionObserver, FLTPaymentQueueHandlerProtocol>
+
+// Stubs
+@property(nonatomic, copy, nullable) BOOL (^addPaymentStub)(SKPayment *payment);
+@property(nonatomic, copy, nullable) void (^showPriceConsentIfNeededStub)(void);
+@property(nonatomic, copy, nullable) void (^stopObservingPaymentQueueStub)(void);
+@property(nonatomic, copy, nullable) void (^startObservingPaymentQueueStub)(void);
+@property(nonatomic, copy, nullable) void (^presentCodeRedemptionSheetStub)(void);
+@property(nonatomic, copy, nullable) void (^restoreTransactions)(NSString *);
+@property(nonatomic, copy, nullable)
+    NSArray<SKPaymentTransaction *> * (^getUnfinishedTransactionsStub)(void);
+@property(nonatomic, copy, nullable) void (^finishTransactionStub)(SKPaymentTransaction *);
+@property(nonatomic, copy, nullable) void (^paymentQueueUpdatedTransactionsStub)
+    (SKPaymentQueue *, NSArray<SKPaymentTransaction *> *);
+
+@end
+
+// An interface representing a stubbed DefaultRequestHandler
+@interface RequestHandlerStub : NSObject <FLTRequestHandlerProtocol>
+
+// Stubs
+@property(nonatomic, copy, nullable) void (^startProductRequestWithCompletionHandlerStub)
+    (ProductRequestCompletion);
+
+@end
+
+#if TARGET_OS_IOS
+@interface FlutterPluginRegistrarStub : NSObject <FlutterPluginRegistrar>
```

Comment:
it's worth checking if it's possible to just use the real registrar in other tests.